instrumentation/akka-http-10.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/akkahttp/client/HttpExtClientInstrumentation.java

```diff
@@ -56,15 +57,14 @@ public static void methodEnter(
       with way of continuing to reusing it.
        */
       Context parentContext = currentContext();
-      if (!tracer().shouldStartSpan(parentContext)) {
+      if (request == null || !instrumenter().shouldStart(parentContext, request)) {
         return;
       }
 
-      // Request is immutable, so we have to assign new value once we update headers
-      AkkaHttpHeaders headers = new AkkaHttpHeaders(request);
-      context = tracer().startSpan(parentContext, request, headers);
+      context = instrumenter().start(parentContext, request);
       scope = context.makeCurrent();
-      request = headers.getRequest();
+      // Request is immutable, so we have to assign new value once we update headers
+      request = setter().inject(request);
```

Comment:
can this be done before calling `start`? just so its the same request instance passed to both `start` and `end`

Reply:
oh, you are so right! 👍